src/sot/sot.cpp

```diff
@@ -471,17 +471,19 @@ computeControlLaw( dynamicgraph::Vector& control,const int& iterTime )
   const Matrix &K = constraintSOUT(iterTime);
   const Matrix::Index mJ = K.cols(); // number dofs - number constraints
 
-  try {
-    control = q0SIN( iterTime );
-    sotDEBUG(15) << "initial velocity q0 = " << control << endl;
-    if( mJ!=control.size() ) { control.resize( mJ ); control.setZero(); }
-  }
-  catch (...)
-    {
-      if( mJ!=control.size() ) { control.resize( mJ );}
-      control.setZero();
-      sotDEBUG(25) << "No initial velocity." <<endl;
+  if (q0SIN.isPlugged()) {
+    try {
+      control = q0SIN( iterTime );
+      sotDEBUG(15) << "initial velocity q0 = " << control << endl;
+      if( mJ!=control.size() )
+        control = Vector::Zero(mJ);
+    } catch (...) { // This catch is kept for backward compatibility.
```

Comment:
Why for backward compatibility? What exception could be thrown here?

Reply:
I don't know what exception was expected to be caught here.
I would favor removing the catch or, at least, catching a given type of exception.

Reply:
I think the catch can be removed. Looking at the code, it seems to me that the rationale was: try to get `q0SIN`; if it is not plugged, this will fail and raise an exception, in which case set `q0` to zero. This is a common idiom in Python ("easier to ask for forgiveness than permission") but it is usually avoided in C++ for performance reasons.
Now that you preliminarily check whether `q0SIN` is plugged, the catch becomes useless I think.

Reply:
I too think that the catch was here for the *not-plugged* exception but I cannot be sure that some piece of code does not use it in another way.
@olivier-stasse do you know if any piece of code in SoT relies **specifically on this exception catching** ?

Reply:
The overall system allowing to perform dynamical casting in the SoT is realized through exception catching.

Reply:
Those exceptions are thrown when plugging signals together. It shouldn't happen at this place.

Reply:
After looking a bit more to the code this was just to make sure that the system has at least a default control value specified by the user. Here you are allowing that this default value to not be set. I think that this is dangerous for newcomer who are not populating the solver. I agree that for advanced users this is annoying. I would therefore suggest that if not task are present and the initial signal is not set then an exception is thrown.